### DISCUSSION TITLE
import/export preferences

### DISCUSSION THREAD

**Discussion by travelgirl-fics** [Able-Bear-Software/tab-nodes-tree#42] [Ideas / Feature Request]
looked through options and did not find an obvious method for saving/exporting and/or importing preferences .  did i miss something, or is this a feature that is on the drawing board?  thanks...

fyi - first time user, and (so far) i like what i see...  hope that enjoyment continues into the future :)

**Comment by zeroarst** [comment]
Hey @travelgirl-fics,
Did you mean exporting all settings, including dark/light style settings and customized themes? I’ve thought about that, but it’s not currently on the roadmap. I’d love to hear more about your use case for this feature. Are you thinking of using it for multiple profiles or maybe after a reinstall? I’m keen to understand how it would help you!

Thanks for the feedback, and I’m glad to hear you’re enjoying the extension so far!

**Comment by travelgirl-fics** [comment]
i hadn't thought of using it for multiple profiles, but i could see where
that might be useful.  i was more interested in having a backup for moving
the app from machine to machine, browser to browser, through reinstalls
and/or borked hard drives, etc...  having preferences backed up somewhere
makes the next install/reinstall just that much faster/cleaner, with no
guessing as to what your preferences might have been last week / last
year...  :)

preferences to include anything changeable/selectable by the app:
dark/lught, customised themes, size/location of option dialogue, number of
nodes, names of nodes, children/parents, etc...  if it can be changed
(whether i know it or not today), yes, please please please :) LOL

thanks <3

**Reply by zeroarst** [reply]
I can implement exporting/importing all settings, including dark/light style settings and customized themes, but not node ones, children/parent relations.  
The tricky part might be compatibilities between new/update settings and old settings. 
Created a ticket for it: #42

Thank you for the feedback.

**Reply by SteveOnorato** [reply]
I got a new laptop and wanted to transfer my customizations over.  Export/import would have saved me some time here.  (Even better is if the extension could leverage Chrome sync to have my customizations already waiting for me, since I logged in to Chrome with the same Google account.)

**Comment by travelgirl-fics** [comment]
excuse me for wondering, but as someone who used to code before retirement,
it seems to me the nodes and their relationships should be simple to note
(if not to reconstruct), even if only done as a hierarchy of text URLs
using a json backbone...  are there difficulties in keeping track of the
nodes, or is it more that it's not simple to keep track of nodes you may
not have created?

**Reply by zeroarst** [reply]
Currently the extension save your node relationships and restore them when restroing your Chrome sessions. You don't need to export those data and restore by youself. If Chrome does not restore your previous session (opened tabs), maybe check if you enable this setting.
![image](https://github.com/user-attachments/assets/aaf57dc6-1a7e-4733-8f9c-94bda78800c0)

**Comment by travelgirl-fics** [comment]
ahh.  understood.  thanks :)
